compiler/cli/cli-common/src/org/jetbrains/kotlin/utils/parametersMap.kt

```diff
@@ -218,10 +219,31 @@ private class AnyArgsConverter : ArgsConverter<Any> {
                 else -> null
             }
 
-        if (value::class.isSubclassOf(parameter.type.jvmErasure)) return ArgsConverter.Result.Success(value)
+        fun evaluateValue(arg: Any): Any? {
+            if (arg::class.isSubclassOf(parameter.type.jvmErasure)) return arg
+            return convertPrimitivesArray(parameter.type, arg)
+        }
+
+        evaluateValue(value)?.let { return ArgsConverter.Result.Success(it) }
+
+        // Handle special scenario where [arg::class] is Array<Any> or Array<*> but
+        // but it's values all contain the proper values
+        val arrayComponentType = parameter
+            .type
+            .classifier
+            .safeAs<KClass<*>>()
+            ?.takeIf { it.java.isArray }
+            ?.java
+            ?.componentType
+            ?.kotlin
 
-        return convertPrimitivesArray(parameter.type, value)?.let { ArgsConverter.Result.Success(it) }
-            ?: ArgsConverter.Result.Failure
+        if (value is Array<*> && arrayComponentType != null) {
+            // TODO: Idea! Maybe we should check if the values in the array are compatible with [arrayComponentType]
+            //  if they aren't perhaps we should fail silently
```

Comment:
Yep, in general, more reporting from this code could be nice. But on the other hand, it partially reimplements the logic already present in the compiler, so ideally we'd like to reuse it, instead of extending this one.